src/components/PositionedFauxCaretWrapper.tsx

```diff
@@ -0,0 +1,78 @@
+import { PropsWithChildren, useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { css } from '../../styled-system/css'
+import Path from '../@types/Path'
+import { isSafari, isTouch } from '../browser'
+import { getBoundingClientRect } from '../device/selection'
+import attributeEquals from '../selectors/attributeEquals'
+import editingValueStore from '../stores/editingValue'
+import equalPath from '../util/equalPath'
+import head from '../util/head'
+
+type Props = PropsWithChildren<{
+  path: Path
+  wrapperElement: HTMLDivElement | null
+}>
+
+/**
+ * Wrapper for positioned faux caret.
+ *
+ * When text within an Editable text node is selected, the faux caret needs to move to the position where
+ * the selection occurred. This differs from what happens when an element node is selected through the use
+ * of commands (cursor down, undo/redo, etc.) where another faux caret is shown/hidden in a static position.
+ *
+ * See FauxCaret.tsx for more information.
+ */
+const PositionedFauxCaretWrapper = ({ children, path, wrapperElement }: Props) => {
+  const editing = useSelector(state => state.editing)
+  const isCursor = useSelector(state => equalPath(path, state.cursor))
+  const isTableCol1 = useSelector(state => attributeEquals(state, head(path), '=view', 'Table'))
+  const [styles, setStyles] = useState({})
```

Comment:
This will need to be explicitly typed since Typescript cannot infer the type from `{}`.